src/util/nlp-time-filter.js

```diff
@@ -0,0 +1,36 @@
+import chrono from 'chrono-node'
+
+const BEFORE_REGEX = /before:"(.*?)"/
+const AFTER_REGEX = /after:"(.*?)"/
+
+// Takes in query as a string and extracts startDate and endDate
+export default function extractTimeFiltersFromQuery(query) {
+    const matchedBefore = query.match(BEFORE_REGEX)
+    const matchedAfter = query.match(AFTER_REGEX)
+
+    let startDate
+    let endDate
+    if (matchedBefore && matchedBefore[1]) {
+        const parsedDate = chrono.parseDate(matchedBefore[1])
+        endDate = parsedDate && parsedDate.getTime()
+    }
+    if (matchedAfter && matchedAfter[1]) {
+        const parsedDate = chrono.parseDate(matchedAfter[1])
+        startDate = parsedDate && parsedDate.getTime()
+    }
+
+    // Checks which comes first before/after and based on that decide where query should be sliced
+    const firstIndexOfTimeFilter = Math.min(
```

Comment:
I wonder if the regex doesn't have a way to to this automatically, something like 
```
extractedQuery.replace(BEFORE_REGEX,'').replace(AFTER_REGEX,'').trim()
```
Would it work?